test/test_ssl.rb

```diff
@@ -1222,6 +1222,56 @@ def test_get_ephemeral_key
     end
   end
 
+  def test_fallback_scsv
+    ctx_proc = proc { |ctx|
+        ctx.ssl_version = :TLSv1_2
+    }
+    start_server(ctx_proc: ctx_proc) do |port|
+      ctx = OpenSSL::SSL::SSLContext.new fallback_scsv: false
+      ctx.ssl_version = :TLSv1_2
+      # Here is OK
+      # TLS1.2 supported and this is what we ask the first time
+      server_connect(port, ctx)
+    end
+
+    ctx_proc = proc { |ctx|
+      ctx.ssl_version = :TLSv1_1
+    }
+    start_server(ctx_proc: ctx_proc) do |port|
+      ctx = OpenSSL::SSL::SSLContext.new fallback_scsv: true
+      ctx.ssl_version = :TLSv1_1
+      # Here is OK too
+      # TLS1.2 not supported, fallback to TLS1.1 and signaling the fallback
+      # Server doesn't support better, so connection OK
+      server_connect(port, ctx)
+    end
+
+    # Here is not OK
+    # TLS1.2 is supported, fallback to TLS1.1 (downgrade attack) and signaling the fallback
+    # Server support better, so refuse the connection
+    sock1, sock2 = socketpair
+    ctx1 = OpenSSL::SSL::SSLContext.new
+    ctx1.min_version = OpenSSL::SSL::TLS1_1_VERSION
+    ctx1.max_version = OpenSSL::SSL::TLS1_2_VERSION
```

Comment:
The server doesn't need to set neither min_version nor max_version since we know all protocol versions are enabled by default.